R/build-home-index.R

```diff
@@ -189,26 +189,35 @@ sidebar_section <- function(heading, bullets, class = make_slug(heading)) {
   )
 }
 
-#' @importFrom memoise memoise
-NULL
-
-cran_link <- memoise(function(pkg) {
+cran_link <- function(pkg) {
   if (!has_internet()) {
     return(NULL)
   }
 
   cran_url <- paste0("https://cloud.r-project.org/package=", pkg)
-
-  if (!httr::http_error(cran_url)) {
+  req <- httr2::request(cran_url)
+  req <- httr2::req_cache(req, path = http_cache_dir(), max_age = 86400)
+  req <- httr2::req_error(req, function(resp) FALSE)
+  resp <- httr2::req_perform(req)
+  if (!httr2::resp_is_error(resp)) {
     return(list(repo = "CRAN", url = cran_url))
   }
 
   # bioconductor always returns a 200 status, redirecting to /removed-packages/
   bioc_url <- paste0("https://www.bioconductor.org/packages/", pkg)
-  req <- httr::RETRY("HEAD", bioc_url, quiet = TRUE)
-  if (!httr::http_error(req) && !grepl("removed-packages", req$url)) {
+  req <- httr2::request(bioc_url)
+  req <- httr2::req_cache(req, path = http_cache_dir(), max_age = 86400)
+  req <- httr2::req_error(req, function(resp) FALSE)
```

Comment:
pkgdown imports magrittr but I suppose not having req as a pipeline is a preference? I was a bit surprised by the repeated explicit assignments.

Reply:
(here and in the other calls to httr2 functions)

Reply:
Using a pipe inside a package just feels weird to me. I might eliminate the relatively few remaining uses in pkgdown, since that would allow us to drop another dependency (which is useful since we're skating very close to the limit of 22)